### PR TITLE
add vlookup transform

### PR DESCRIPTION
**Pull request by cpdough**
we <3 excel

### CODE REVIEW

File: rasgotransforms/rasgotransforms/transforms/vlookup/vlookup.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: vlookup
+tags:
+  - table
+  - join
+  - reshape
+description: |
+  ## Inspired by Excel... a VLookup experience that works in SQL
+
+  ### Required Inputs
+  - Lookup Column: The column to look up in the Lookup Table. Make sure the column is named the same in both tables.
```
Comment (griffatrasgo) [inline rasgotransforms/rasgotransforms/transforms/vlookup/vlookup.yaml:10]: How soon do you anticipate users wanting to join on columns that don't have the same name? i.e. should we loosen the same name rule and allows for anchor_ & lookup_column args in the event they aren't named the same (which excel supports)? Or do we just push users to a regular join transform at that point?

File: rasgotransforms/rasgotransforms/transforms/vlookup/vlookup.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{%- else -%}
+    {%- set lookup_table_cols = get_columns(lookup_table) -%}
+{%- endif -%}
+{%- set lookup_table_name = table_from_fqtn(lookup_table) -%}
+
+
+SELECT base.*, 
+{%- for column in lookup_table_cols %}
+    {%- if column in source_col_names -%}
+        lookup.{{ column }} as {{ lookup_table_name }}_{{ column }}{{ ', ' if not loop.last }}
```
Comment (griffatrasgo) [inline rasgotransforms/rasgotransforms/transforms/vlookup/vlookup.sql:20]: nice touch 👍

**Review by griffatrasgo — APPROVED** [review]
Nice transform. Good kickoff to Rasgo's "Excel phase" :)
As a dumb user, I could see value in supporting different column names across the tables, but i defer to the author

File: rasgotransforms/rasgotransforms/transforms/vlookup/vlookup.yaml (hunk 1 above)
Comment (cpdough) [reply rasgotransforms/rasgotransforms/transforms/vlookup/vlookup.yaml:10]: yeah i was wondering the same thing, but i tried to stay true to how vlookup does the lookup and it doesn't have a concept of "mapping" one column name to another (it just goes by position)

**Pull request merged**